packages/core/test/stylable-import-tools.spec.ts

```diff
@@ -0,0 +1,466 @@
+import { expect } from 'chai';
+import { parse } from 'postcss';
+import { Diagnostics, ensureStylableImports } from '@stylable/core';
+import { ensureImportsMessages } from '@stylable/core/dist/stylable-imports-tools';
+
+describe('ensureStylableImports', () => {
+    describe('all modes', () => {
+        it('should not modify when no importPatches provided', () => {
+            const root = parse(`
+                :import {-st-from: 'x'; -st-default: A }
+            `);
+
+            const diag = new Diagnostics();
+
+            ensureStylableImports(root, [], { mode: 'patch-only' }, diag);
+            ensureStylableImports(root, [], { mode: ':import' }, diag);
+            ensureStylableImports(root, [], { mode: 'st-import' }, diag);
+
+            const importNode = root.nodes[0];
+            expect(diag.reports, 'No diagnostics').to.have.lengthOf(0);
+            expect(importNode.toString()).to.equal(`:import {-st-from: 'x'; -st-default: A }`);
+        });
+    });
+
+    describe('patch-only mode', () => {
+        it('should not add missing imports in patch-only mode', () => {
+            const root = parse(``);
+
+            const diag = new Diagnostics();
+
+            ensureStylableImports(
+                root,
+                [{ request: 'missing', defaultExport: 'Missing' }],
+                { mode: 'patch-only' },
+                diag
+            );
+
+            expect(diag.reports, 'No diagnostics').to.have.lengthOf(0);
+            expect(root.nodes, 'no imports added').to.have.lengthOf(0);
+        });
+
+        it('should add default', () => {
+            const root = parse(`
+                :import {-st-from: 'x'; }
+            `);
+
+            const diag = new Diagnostics();
+
+            ensureStylableImports(
+                root,
+                [{ request: 'x', defaultExport: 'Test' }],
+                { mode: 'patch-only' },
+                diag
+            );
+
+            const importNode = root.nodes[0];
+            expect(diag.reports, 'No diagnostics').to.have.lengthOf(0);
+            expect(importNode.toString()).to.equal(`:import {-st-from: 'x';-st-default: Test; }`);
+        });
+
+        it('should add named import', () => {
+            const root = parse(`
+                :import {-st-from: 'x'; }
```

Comment:
If you had 2 imports from the same `from` path, would it add the import symbols to both and cause a duplicate symbol error?

Reply:
fixed